cmd/src/gateway_benchmark.go

```diff
@@ -152,7 +178,30 @@ func benchmarkEndpoint(client *http.Client, url string) time.Duration {
 		fmt.Printf("Error reading response body: %v\n", err)
 		return 0
 	}
+	if resp.StatusCode != http.StatusOK {
+		fmt.Printf("non-200 response: %v\n", resp.Status)
+		return 0
+	}
+
+	return time.Since(start)
+}
 
+func benchmarkEndpointWebSocket(conn *websocket.Conn) time.Duration {
+	start := time.Now()
+	err := conn.WriteMessage(websocket.TextMessage, []byte("ping"))
+	if err != nil {
+		fmt.Printf("WebSocket write error: %v\n", err)
+		return 0
+	}
+	_, message, err := conn.ReadMessage()
+	if err != nil {
+		fmt.Printf("WebSocket read error: %v\n", err)
+		return 0
+	}
+	if string(message) != "pong" {
```

Comment:
I've changed my other PR https://github.com/sourcegraph/sourcegraph/pull/1526 to respond with "pong" on all three endpoints.

Reply:
Fantastic!